lua/obsidian/command.lua

```diff
@@ -216,6 +216,123 @@ command.search = function(client, data)
   end
 end
 
+--- Insert a template
+---
+---@param client obsidian.Client
+---@param data table
+command.insert_template = function(client, data)
+  if not client.opts.templates.subdir then
+    echo.err "No templates folder defined in setup()"
+    return
+  end
+
+  local templates_dir = Path:new(client.dir) / client.opts.templates.subdir
+  if not templates_dir:is_dir() then
+    echo.err(string.format("%s is not a valid directory for templates", templates_dir))
+    return
+  end
+
+  -- We need to get these upfront otherwise
+  -- Telescope hijacks the current window
+  local buf = vim.api.nvim_win_get_buf(0)
+  local win = vim.api.nvim_get_current_win()
+  local row, _ = unpack(vim.api.nvim_win_get_cursor(win))
+
+  local apply_template = function(name)
+    local template_path = Path:new(templates_dir / name)
+    local date_format = client.opts.templates.date_format or "%Y-%m-%d"
+    local time_format = client.opts.templates.time_format or "%H:%M"
+    local date = tostring(os.date(date_format))
+    local time = tostring(os.date(time_format))
+    local current_fp = vim.api.nvim_buf_get_name(buf)
+    local _, _, title = string.find(vim.fs.normalize(current_fp), ".*/(.-)%.md")
+    title = title or vim.fs.basename(current_fp)
```

Comment:
It would be more robust to use the `Note` class here:
```suggestion
    local title = Note.from_buffer(buf, client.dir):display_name()
```